example-app/src/main/java/com/adyen/checkout/example/ui/card/compose/SessionsCardScreen.kt

```diff
@@ -0,0 +1,142 @@
+import androidx.compose.foundation.layout.ime
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.windowInsetsPadding
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.ArrowBack
+import androidx.compose.material3.CircularProgressIndicator
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.material3.TopAppBar
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.hilt.navigation.compose.hiltViewModel
+import com.adyen.checkout.card.CardComponent
+import com.adyen.checkout.card.CardConfiguration
+import com.adyen.checkout.components.compose.AdyenComponent
+import com.adyen.checkout.components.compose.get
+import com.adyen.checkout.components.core.action.Action
+import com.adyen.checkout.example.ui.card.SessionsCardComponentData
+import com.adyen.checkout.example.ui.card.SessionsCardViewModel
+import com.adyen.checkout.example.ui.compose.ResultContent
+
+@Composable
+internal fun SessionsCardScreen(
+    onBackPressed: () -> Unit,
+) {
+    Scaffold(
+        modifier = Modifier.windowInsetsPadding(WindowInsets.ime),
+        topBar = {
+            TopAppBar(
+                title = { Text(text = "Card component with sessions") },
+                navigationIcon = {
+                    IconButton(onClick = { onBackPressed() }) {
+                        Icon(Icons.Filled.ArrowBack, contentDescription = "Back")
+                    }
+                },
+            )
+        },
+    ) { innerPadding ->
+        MainContent(Modifier.padding(innerPadding))
+    }
+}
+
+@Suppress("DestructuringDeclarationWithTooManyEntries")
+@Composable
+private fun MainContent(
+    modifier: Modifier = Modifier,
+    viewModel: SessionsCardViewModel = hiltViewModel(),
+) {
+    Box(
+        modifier = modifier.fillMaxSize(),
+        contentAlignment = Alignment.Center,
+    ) {
+        val uiState by viewModel.uiState.collectAsState()
+
+        val (cardConfiguration, isLoading, toastMessage, componentData, action, finalResult) = uiState
+
+        if (isLoading) {
+            CircularProgressIndicator()
+        }
+
+        if (toastMessage != null) {
+            val context = LocalContext.current
+            LaunchedEffect(toastMessage) {
+                Toast.makeText(context, toastMessage, Toast.LENGTH_SHORT).show()
+                viewModel.toastMessageConsumed()
```

Comment:
I would suggest creating a function argument, instead of directly calling a method on viewMode here.
E.g. onTastMessageConsumed: () -> Unit as an argument for MainContent and in SessionsCardScreen I would just pass viewModel:: toastMessageConsumed to the MainContent.

Reply:
Same could work for viewModel::actionConsumed